src/plugins/intel_cpu/src/emitters/tpp/x64/jit_equation_emitter.cpp

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "jit_equation_emitter.hpp"
+#include "transformations/tpp/x64/op/equation.hpp"
+
+using namespace Xbyak;
+using namespace dnnl::impl;
+using namespace dnnl::impl::cpu::x64;
+
+namespace ov {
+namespace intel_cpu {
+using jit_generator = dnnl::impl::cpu::x64::jit_generator;
+using cpu_isa_t = dnnl::impl::cpu::x64::cpu_isa_t;
+using ExpressionPtr = ov::snippets::lowered::ExpressionPtr;
+
+EquationTppEmitter::EquationTppEmitter(jit_generator* h, cpu_isa_t isa, const ExpressionPtr& expr) :
+                                       TppEmitter(h, isa, expr), m_num_inputs(expr->get_input_count()) {
+    const auto& eq_tpp = ov::as_type_ptr<tpp::op::EquationTPP>(expr->get_node());
+    OV_CPU_JIT_EMITTER_ASSERT(eq_tpp, "Invalid TPP node type detected");
+    auto get_MN = [this](int arg_idx){
+        const auto& subtensor = get_projected_subtensor(io_port_descriptors[arg_idx]);
+        OV_CPU_JIT_EMITTER_ASSERT(subtensor.size() == 2, "TPP supports only 2D subtensors");
+        return std::make_pair(static_cast<libxsmm_blasint>(*++subtensor.rbegin()),
+                              static_cast<libxsmm_blasint>(*subtensor.rbegin()));
+    };
+
+    m_equation_id = libxsmm_meqn_create();
+    const auto op_metadata = libxsmm_create_meqn_op_metadata(m_equation_id, -1);
+    const auto sing_attr = libxsmm_create_matrix_arg_attributes(LIBXSMM_MATRIX_ARG_TYPE_SINGULAR, LIBXSMM_MATRIX_ARG_SET_TYPE_NONE, 0, 0);
+    libxsmm_blasint M, N;
+    for (const auto& op_desc : eq_tpp->get_op_descs()) {
+        switch (op_desc.get_arity()) {
+            case tpp::op::OpDescTPP::ARITY::BINARY: {
+                auto flags = op_desc.get_flags();
+                libxsmm_meqn_push_back_binary_op(op_metadata, op_desc, exec_dtype, flags);
+                break;
+            } case tpp::op::OpDescTPP::ARITY::UNARY: {
+                libxsmm_meqn_push_back_unary_op(op_metadata, op_desc, exec_dtype, LIBXSMM_MELTW_FLAG_UNARY_NONE);
+                break;
+            } case tpp::op::OpDescTPP::ARITY::ZERO: {
+                const auto arg_idx = static_cast<int>(op_desc);
+                std::tie(M, N) = get_MN(arg_idx);
+                auto metadata = libxsmm_create_meqn_arg_metadata(m_equation_id, arg_idx);
+                auto shape = libxsmm_create_meqn_arg_shape(N, M,
+                                                           static_cast<libxsmm_blasint>(io_strides[arg_idx]),
+                                                           io_dtypes[arg_idx]);
+                OV_CPU_JIT_EMITTER_ASSERT(libxsmm_meqn_push_back_arg(metadata, shape, sing_attr) == 0,
+                                          "Failed to push back arg to tpp equation");
+                break;
+            }
+            default:
+                OV_CPU_JIT_EMITTER_THROW("Unhandled tpp::op::OpDescTPP::ARITY");
+        }
+    }
+
+    libxsmm_meqn_tree_print(m_equation_id);
+    libxsmm_meqn_rpn_print(m_equation_id);
```

Comment:
Are these `*_print` calls needed? The name suggests that this might be debug functions

Reply:
Yeah, I left them on purpose, because otherwise Snippets developers wouldn't know about this libxsmm funcional (it's not documented).
On the other hand you are right, probably we shouldn't print every time in production (even though it's an experimental feature.
So what do you think about this intermediate solution? I left explanatory comment and commented these lines to facilitate future debugging.

Reply:
The intermediate solution is good enough